documentation/src/docs/basic-reader-and-writer.md

```diff
@@ -263,130 +224,102 @@ public void run() {
+of one or more Readers that consume from a Stream in parallel. Before we create
+a Reader, we need to either create a `ReaderGroup` (or be aware of the name of an
+existing `ReaderGroup`). This application only uses the basics from Reader Group.
+
+`ReaderGroup` objects are created from a `ReaderGroupManager` object. The `ReaderGroupManager` object, in turn, is created on a given Scope with a URI to one of the Pravega Controllers, very much
+like a `ClientFactory` is created. Note that, the `createReaderGroup()` is also in a try-with-resources statement to make sure that the `ReaderGroupManager` is properly cleaned up. The `ReaderGroupManager` allows a developer to create, delete and retrieve `ReaderGroup` objects using the name.
+
+To create a `ReaderGroup`, the developer needs a name for the Reader Group, a
+configuration with a set of one or more Streams to read from. The Reader Group's name (alphanumeric) might be meaningful to the application, like
+"WebClickStreamReaders".  In cases where we require multiple Readers reading in parallel and each Reader in a separate
+process, it is helpful to have a human readable name for the Reader Group. In this example, we have one Reader, reading in isolation, so a UUID is a safe way to
+name the Reader Group. The Reader Group is created via the
+`ReaderGroupManager` and since the `ReaderGroupManager` is created within the
+context of a Scope, we can safely conclude that Reader Group names are namespaced
 by that Scope.  
 
-The ReaderGroupConfig right now doesn't have much behavior.  The developer
-specifies the Stream which should be part of the ReaderGroup and its lower and
-upper bounds. In our case, on line 11, we start at the beginning of the Stream.
-Other configuration items, such as specifying checkpointing etc. are options
-that will be available through the ReaderGroupConfig.  But for now, we keep it
-simple.
+The developer specifies the Stream which should be the part of the Reader Group and its lower and
+upper bounds. In the sample application, we start at the beginning of the Stream as follows:
 
-The fact that a ReaderGroup can be configured to read from multiple Streams is
-kind of cool.  Imagine a situation where I have a collection of Stream of sensor
-data coming from a factory floor, each machine has its own Stream of sensor
-data.  I can build applications that use a ReaderGroup per Stream so that the
-app reasons about data from exactly one machine.  I can build other apps that
-use a ReaderGroup configured to read from all of the Streams.  In our case, on
-line 14, the ReaderGroup only reads from one Stream.
+```Java
+final ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
+                                                             .stream(Stream.of(scope, streamName))
+                                                             .build();
+```
+Other configuration items, such as specifying checkpointing etc., are options
```

Comment:
Maybe, we have to point to the Checkpoint concept:
`Other configuration items, such as specifying checkpointing etc.,` -> Other configuration items, such as [Checkpointing](http://pravega.io/docs/latest/pravega-concepts/#readergroup-checkpoints), are options...